pyfarm/master/api/tasklogs.py

```diff
@@ -302,14 +303,18 @@ def get(self, job_id, task_id, attempt, log_identifier):
             logfile = open(path, "rb")
             return send_file(logfile)
         except IOError:
-            agent = log.agent
-            if not agent:
-                return (jsonify(
-                    path=path, log=log_identifier,
-                    error="Logfile is not available on master and agent is "
-                          "not known"), NOT_FOUND)
-            return redirect(agent.api_url() + "/task_logs/" + log_identifier,
-                            TEMPORARY_REDIRECT)
+            try:
+                compressed_logfile = GzipFile("%s.gz" % path, "rb")
+                return Response(compressed_logfile.read(), mimetype="text/csv")
```

Comment:
Probably should build a streaming response here so we don't load everything into memory first.

http://flask.pocoo.org/docs/0.10/patterns/streaming/#basic-usage